pkg/auth/dependency/newinteraction/nodes/generate_recovery_code_begin.go

```diff
@@ -0,0 +1,60 @@
+package nodes
+
+import (
+	"github.com/authgear/authgear-server/pkg/auth/dependency/authenticator"
+	"github.com/authgear/authgear-server/pkg/auth/dependency/newinteraction"
+)
+
+func init() {
+	newinteraction.RegisterNode(&NodeGenerateRecoveryCodeBegin{})
+}
+
+type EdgeGenerateRecoveryCodeBegin struct{}
```

Comment:
It's better to name it `EdgeGenerateRecoveryCode`, since the edge can go to both Begin & End node.